.github/workflows/ci.yml

```diff
@@ -531,3 +549,86 @@ jobs:
           path: tests/test_docs/diffs/*.jpeg
         # Always run this step to publish test results, even on failures
         if: ${{ always() }}
+
+  end-to-end-test:
+    name: "Sandbox upgrade E2E test"
+    runs-on: ubuntu-latest
+    needs:
+      - build-new-container-image
+      - build-container-image
```

Comment:
What happens if `build-new-container-image` ends quicker than `buld-container-image`?

Reply:
Whoops, I hadn't seen this, sorry!